packages/server-core/src/cluster/logs-api/logs-api.class.ts

```diff
@@ -50,16 +51,23 @@ export class LogsApiService implements ServiceInterface<void, any, LogsApiParams
 
       if (Array.isArray(data)) {
         for (const item of data) {
-          this._processLogItem(item, userId)
+          await this._processLogItem(item, userId)
         }
       } else {
-        this._processLogItem(data, userId)
+        await this._processLogItem(data, userId)
       }
     }
   }
 
-  _processLogItem = (logItem, userId?: string) => {
-    const { msg, level } = logItem
+  _processLogItem = async (logItem, userId?: string) => {
+    const { msg, level, action } = logItem
+
+    delete logItem.action
+
+    if (action === 'analytics') {
+      await logToBigQuery({ ...logItem })
+      return
+    }
```

Comment:
@infinitelos @barankyle should we add following check in below if condition so that it logs to BQ only when its enabled and there is a fallback to old analytics logging approach if BQ is not setup?

```suggestion
    if (action === 'analytics' && process.env.BQ_PROJECT_ID && process.env.BQ_DATASET_ID && process.env.BQ_TABLE_ID) {
      await logToBigQuery({ ...logItem })
      return
    }
```

Reply:
It can't hurt.